e2etest/contract_staking_test.go

```diff
@@ -1715,6 +1715,62 @@ func TestContractStaking(t *testing.T) {
 		r.True(ok)
 		r.EqualValues(identityset.Address(delegateIdx).String(), bt.Candidate.String())
 	})
+
+	t.Run("transfer token", func(t *testing.T) {
```

Comment:
if this test case is added in first place, it should've caught this bug, is that right?
like we discussed, need to add comprehensive tests, not to cover every line of code, but all newly added funcs/APIs should be covered